cmd/cli/version/version.go

```diff
@@ -104,13 +107,71 @@ func (oV *VersionOptions) Run(ctx context.Context, cmd *cobra.Command) error {
 	if !oV.ClientOnly {
 		serverVersion, err := util.GetAPIClient(ctx).Version(ctx)
 		if err != nil {
-			log.Ctx(ctx).Error().Err(err).Msgf("could not get server version")
-			return err
+			return fmt.Errorf("error running version: %w", err)
+		} else {
+			versions.ServerVersion = serverVersion
+			columns = append(columns, serverVersionColumn)
 		}
+	}
+
+	// Add additional information
+	versions.OperatingSystem = runtime.GOOS
+	versions.Architecture = runtime.GOARCH
+	versions.UserID = "some-user-id" // Replace this with your method to get UserID
 
-		versions.ServerVersion = serverVersion
-		columns = append(columns, serverVersionColumn)
+	var serverVersion string
+	if versions.ServerVersion != nil {
+		serverVersion = versions.ServerVersion.GitVersion
 	}
 
+	checkForUpdates(ctx, versions.ClientVersion.GitVersion, serverVersion, versions.OperatingSystem, versions.Architecture, versions.UserID)
+
 	return output.OutputOne(cmd, columns, oV.OutputOpts, versions)
 }
+
+type ServerResponse struct {
+	Version string `json:"version"`
+	Message string `json:"message"`
+}
+
+func checkForUpdates(ctx context.Context, currentClientVersion, currentServerVersion, os, arch, userID string) {
+	u, err := url.Parse("http://update.bacalhau.org/version")
+	if err != nil {
+		log.Ctx(ctx).Error().Err(err).Msg("Failed to parse URL.")
+		return
+	}
+
+	q := u.Query()
+	q.Set("clientVersion", currentClientVersion)
+	if currentServerVersion != "" {
+		q.Set("serverVersion", currentServerVersion)
+	}
+	// Add OS, Architecture, and UserID to the query parameters
+	q.Set("operatingSystem", os)
+	q.Set("architecture", arch)
+	q.Set("userID", userID)
+	
+	u.RawQuery = q.Encode()
+
+	resp, err := http.Get(u.String())
+	if err != nil {
+		log.Ctx(ctx).Error().Err(err).Msg("Failed to fetch the latest version from the server.")
+		return
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		log.Ctx(ctx).Error().Err(err).Msg("Failed to read response body.")
+		return
+	}
+
+	var serverResponse ServerResponse
+	err = json.Unmarshal(body, &serverResponse)
+	if err != nil {
+		log.Ctx(ctx).Error().Err(err).Msg("Failed to unmarshal the server response.")
+		return
+	}
+
+	fmt.Println(serverResponse.Message)
```

Comment:
Rather than print the response here, we should return the `ServerResponse`. This will allow callers to determine how the update message is displayed to the user. e.g. as json, yaml, columns, text, etc.